### PR TITLE
Add clientside code

### PR DESCRIPTION
**Pull request by sioanis**
This PR will add the client-side code that will consume a GraphQL connected database:

- Testing: `@testing-library`
- Automated scripts: `react-scripts` (easier build scripts / could remove this later on)
- Routing: `react-router-dom`
  - `/login` routes to `Login`
  - `/signup` routes to `SignUp`

- Local state management: Apollo client `@apollo/react-hooks`, `apollo-boost`
  - `/resolvers` will handle local state data mutation 
  - `/schema` will define type definitions when local state changes

---
Three new components using `@material-ui/core` - `Login`, `SignUp`, `Copyright`:
Screen Shot 2020-01-18 at 5 53 57 PM
Screen Shot 2020-01-18 at 5 53 53 PM

### CODE REVIEW

File: client/src/serviceWorker.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
+// This optional code is used to register a service worker.
```
Comment (sioanis) [inline client/src/serviceWorker.js]: Will do a spike on service workers in relation to apollo cache. https://developers.google.com/web/fundamentals/primers/service-workers

Comment (michaelyuen) [reply client/src/serviceWorker.js]: 👍 If you want to use Clubhouse then I think it's a good thing to track there.

File: client/src/logo.svg [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 841.9 595.3">
```
Comment (michaelyuen) [inline client/src/logo.svg]: is this being used?

File: client/src/components/SignUp/index.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+import CssBaseline from '@material-ui/core/CssBaseline';
+import TextField from '@material-ui/core/TextField';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Checkbox from '@material-ui/core/Checkbox';
+import { Link } from "react-router-dom";
+import Grid from '@material-ui/core/Grid';
+import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
+import Typography from '@material-ui/core/Typography';
+import { makeStyles } from '@material-ui/core/styles';
+import Container from '@material-ui/core/Container';
```
Comment (michaelyuen) [inline client/src/components/SignUp/index.js]: can the imports be condensed?

File: client/src/components/SignUp/index.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+const useStyles = makeStyles(theme => ({
```
Comment (michaelyuen) [inline client/src/components/SignUp/index.js]: is this the way we have to write styles?

File: client/src/components/Copyright/index.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import Link from '@material-ui/core/Link';
+
+export default function Copyright() {
+  return (
+    <Typography variant="body2" color="textSecondary" align="center">
+      {'Copyright © '}
+      <Link color="inherit" href="https://tripster.vip">
+        Tripster
+      </Link>{' '}
+      {2020}
```
Comment (michaelyuen) [inline client/src/components/Copyright/index.js]: we should make this dynamic now so we don't have to worry later

File: client/README.md [hunk 6]
```diff
@@ -1 +1,46 @@
 # tripster client - built on react
```
Comment (michaelyuen) [inline client/README.md]: add an emoji here 🌮

File: client/src/logo.svg (hunk 2 above)
Comment (sioanis) [reply client/src/logo.svg]: No, I just removed it. Good catch. 👍

File: client/src/App.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
+import { defaults, resolvers } from "./resolvers";
+import { typeDefs } from "./schema";
+import Login from './components/Login';
+import SignUp from './components/SignUp';
+import Copyright from './components/Copyright';
+
+const cache = new InMemoryCache();
+const client = new ApolloClient({
+  uri: 'https://48p1r2roz4.sse.codesandbox.io',
```
Comment (michaelyuen) [inline client/src/App.js]: what's this uri?

File: client/src/App.test.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import App from './App';
+
+test('render', () => {
+  expect(render(<App />));
```
Comment (michaelyuen) [inline client/src/App.test.js]: 🤔

File: client/src/App.js (hunk 7 above)
Comment (sioanis) [reply client/src/App.js]: Test data, let me add a comment here.

**Pull request merged**